### PR TITLE
allow user to adjust ulimits

### PR DESCRIPTION
**Pull request by jedwards4b**
Adds a resource limits section to config_machines.xml allowing the user to adjust environment ulimits

Test suite: scripts_regression_tests.py
Test baseline: 
Test namelist changes: 
Test status: bit for bit
Addresses #771 

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

File: scripts/lib/CIME/XML/env_mach_specific.py [hunk 1]
```diff
@@ -85,6 +85,24 @@ def load_env(self, compiler, debug, mpilib):
...
 
+
+        self._get_resources_for_case(compiler, debug, mpilib)
+
+    def _get_resources_for_case(self, compiler, debug, mpilib):
+        resource_nodes = self.get_nodes("resource_limits")
+        if resource_nodes is not None:
+            nodes = self._compute_resource_actions(resource_nodes, compiler, debug, mpilib)
+            for name, val in nodes:
+                try:
```
Comment (jgfouca) [inline scripts/lib/CIME/XML/env_mach_specific.py]: What's the point of this try: if you're just going to re-raise the exception?

Comment (jedwards4b) [reply scripts/lib/CIME/XML/env_mach_specific.py]: It's a place holder for figuring out something better - I'll remove it.

**Comment by apcraig** [comment]
Can someone provide an example of how to use this for say "limit coredumpsize unlimited".  There is nothing in the documentation about stuff like this.  thanks.

**Comment by jedwards4b** [comment]
-1 is a synonym for unlimited so in config_machines.xml (or in env_mach_specific.xml for a case) you would add
```
<resource_limits>
      <resource name="COREDUMPSIZE">-1</resource>
</resource_limits>
```

**Comment by apcraig** [comment]
Looks like there is a very specific order things have to be in the xml.  Anyway, when it doesn't complain about the order, I get this,

> Traceback (most recent call last):
>   File "./preview_namelists", line 57, in <module>
>     _main_func(__doc__)
>   File "./preview_namelists", line 54, in _main_func
>     case.create_namelists(component=args.component)
>   File "/p/home/apcraig/E3SM-ARRM/E3SM/cime/scripts/Tools/../../scripts/lib/CIME/case/preview_namelists.py", line 59, in create_namelists
>     self.load_env()
>   File "/p/home/apcraig/E3SM-ARRM/E3SM/cime/scripts/Tools/../../scripts/lib/CIME/case/case.py", line 1305, in load_env
>     env_module.load_env(self, job=job, verbose=verbose)
>   File "/p/home/apcraig/E3SM-ARRM/E3SM/cime/scripts/Tools/../../scripts/lib/CIME/XML/env_mach_specific.py", line 84, in load_env
>     self._get_resources_for_case(case)
>   File "/p/home/apcraig/E3SM-ARRM/E3SM/cime/scripts/Tools/../../scripts/lib/CIME/XML/env_mach_specific.py", line 91, in _get_resources_for_case
>     attr = getattr(resource, name)
> AttributeError: 'module' object has no attribute 'COREDUMPSIZE'
> 

I am adding,

```
  <resource_limits>
      <resource name="COREDUMPSIZE">-1</resource>
      <resource name="STACKSIZE">-1</resource>
  </resource_limits>

```
and I am using cime version

> commit da5433476e43393686c42fafa06a61c98f127f8a
> Merge: dafe289a4 eaf53f9b4
> Author: Balwinder Singh <balwindersingh@gmail.com>
> Date:   Fri Dec 14 13:18:58 2018 -0800
> 

I have also tried with lower case "coredumpsize" and "stacksize" and get the same error.

**Pull request merged**